backend/src/main/java/io/dcbn/backend/evidenceFormula/services/DefaultFunctionProvider.java

```diff
@@ -11,9 +14,70 @@
 
 public class DefaultFunctionProvider extends FunctionProvider {
 
-    public DefaultFunctionProvider() {
+
+    private VesselCache vesselCache;
+    private AoiCache aoiCache;
+
+    private Object inArea(List<Object> parameters) {
+        String aoiName = (String) parameters.get(0);
+        if (aoiCache.getAoi(aoiName) == null) {
+            throw new SymbolNotFoundException(aoiName);
+        }
+        AreaOfInterest aoi = aoiCache.getAoi(aoiName);
+        correlatedAois.add(aoi);
+
+        Point vesselPoint = createPoint(currentVessel);
+
+        return aoi.getGeometry().contains(vesselPoint);
+    }
+
+    private Object distanceToNearest(List<Object> parameters) {
+        Set<Vessel> allVessels = vesselCache.getAllVesselsInTimeSlice(currentTimeSlice);
+        return distance(currentVessel, allVessels);
+    }
+
+    private Object distanceToNearestType(List<Object> parameters) {
+        Set<Vessel> allVesselsByType = vesselCache.getAllVesselsInTimeSliceByType(currentTimeSlice, (VesselType) parameters.get(0));
```

Comment:
Die Parameter sind nur Strings, Doubles oder Booleans. Du musst den Parameter mit VesselType.valueOf in ein VesselType umwandeln.